docs/environments/sudoku.md

```diff
@@ -0,0 +1,52 @@
+# Sudoku Environment
+
+<p align="center">
+        <img src="../env_anim/sudoku.gif" width="500"/>
+</p>
+
+We provide here a Jax JIT-able implementation of the
+[Sudoku](https://en.wikipedia.org/wiki/Sudoku) puzzle game.
+
+
+## Observation
+The observation given to the agent consists of:
+
+
+- `board`: jax array (int32) of shape (9,9):
+    empty cells are represented by -1, and filled cells are represented by 0-8.
+- `action_mask`: jax array (bool) of shape (9,9,9):
+    indicates which actions are valid.
+
+
+
+## Action
+The action space is a `MultiDiscreteArray` of integer values representing coordinates of the square
+to explore and the digits to write in the cell, e.g. `[3, 6, 8]` for writing the digit `9` in
+the cell located on the fourth row and seventh column.
+
+
+## Reward
+The reward is `1` at the end of the episode if the board is correctly solved, and `0` in every
+other case.
+
+## Termination
+An episode terminates where there is no more legal actions available, this could happen if the
+board is solved or if the agent finds himself in a dead-end.
```

Comment:
```suggestion
An episode terminates when there are no more legal actions available, this could happen if the
board is solved or if the agent finds itself in a dead-end.
```